src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -8,10 +8,60 @@
 
 namespace GraphQL.Types
 {
+    /// <summary>
+    /// Provides lookup for all schema types and has algorithms for discovering them.
+    /// </summary>
     public class GraphTypesLookup
     {
-        private readonly IDictionary<string, IGraphType> _types = new Dictionary<string, IGraphType>();
+        // Introspection types http://spec.graphql.org/June2018/#sec-Schema-Introspection
+        private readonly Dictionary<Type, IGraphType> _introspectionTypes = new IGraphType[]
```

Comment:
All these types are always added to the schema. Each type is created only **once** within schema instance. Before these changes types were created several times. It is also true for built-in scalars.